assignment-2/assignment2.rb

```diff
@@ -0,0 +1,220 @@
+  puts "The #{room} has been cleaned."
+end
+
+def print_robot_face
+  puts "__^_____^__"
+  puts "|         |"
+  puts "|  ^   ^  |"
+  puts "|    >    |"
+  puts "|    o    |"
+  puts "|_________|"
+end
+
+def print_robot_face_money
+  puts "__^_____^__"
+  puts "|         |"
+  puts "|  ^   ^  |"
+  puts "|    >    |"
+  puts "|    $    |"
+  puts "|_________|"
+end
+
+def clean_house(rooms)
+  puts "Thank you for procuring my service."
+  print_robot_face
+  puts "\n"
+  sleep(1)
+  rooms.each do |room|
+    puts "Begin cleaning #{room}..."
+    clean_room(room)
+    puts "*" * 40
+    puts "\n"
+    sleep(1)
+  end
+
+  puts "House cleaning complete. Please remit payment. Venmo preferred."
+  print_robot_face_money
+end
+
+
+clean_house(rooms)
+=begin
+2. Tiny Instacart!
+Compose a simple terminal program using one or more of the common objects from Instacart (orders, drivers, batches, order_items, payments, etc)
+
+Design a desired workflow. ex. The entire flow of a driver from accepting a batch to delivery
+Using the tools and concepts from the completed room cleaner program, construct a self-running terminal program that successfully runs through your intended workflow.
+Test your program thoroughly to ensure that it runs as expected.
+Possible Extensions:
+
+Ruby offers alternatives to if/else including unless, while, and until.
+Set parameters for your program through user input
+Make sure to incorporate if/else and the appropriate operators.
+Look up how to use a ruby hash to create an array with multiple attributes per array element and run an each loop that checks the values of said attributes. ex. (`drivers = [{name: "Rose", role: "in-store-shopper"}, {name: "Donna", role: "shift-lead"}, {name: "Amelia Pond", role: "full-service"}, {name: "Martha Jones", role: "cashier"})
+=end
+
+def accept_batch
+  puts "You have a new batch, enter \'Yes\' to accept or \'No\' to decline "
+  response = gets.chomp
+  if response.downcase == "yes"
+    puts 'Lets get started!'
+  elsif response.downcase == "no"
+    puts 'You have failed to accept the batch! Please wait for the next batch'
+  else
+    puts 'You have failed to accept the batch! Please wait for the next batch'
+  end
+end
+
+def deliver_batch
+  puts "Deliver the order to 123 Beagle St unit 2006."
+  puts "Enroute"
+  sleep(1)
+  puts "."
+  sleep(1)
+  puts ".."
+  sleep(1)
+  puts "..."
+  puts "You have arrived at the destination."
+end
+
+def customer_receive
+  puts "Knock on the customer's door. Is the customer home? Yes or No"
+  loop do
+    response = gets.chomp
+    if response.downcase == "yes"
+      puts "Great job successfully delivering the batch!"
+      break
+    elsif response.downcase == "no"
+      puts "Leave a message for the customer that the batch has been left at the door"
+      break
+    else
+      puts "Leave a message for the customer that the batch has been left at the door"
+      break
+    end
+  end
+end
+
+
+accept_batch
+deliver_batch
```

Comment:
looks like this will progress to the `deliver_batch` and `customer_receive` methods regardless of whether the shopper accepted the batch. we can have accept_batch return a true or false to determine whether to run the downstream methods

Reply:
Added a loop to only continue if the batch was accepted, and for a new batch to appear if they failed to accept the original.